### PR TITLE
feat: Availability Route added for event tickets

### PR DESCRIPTION
**Pull request by MeghalBisht**
Reqd for : fossasia/open-event-frontend#6269 and fossasia/open-event-frontend#6275

#### Short description of what this resolves:
/events/ID/tickets/availability route created which returns each ticket's remaining count

Response :
kk

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
This shows deleted tickets as well

**Comment by MeghalBisht** [comment]
@iamareebjamal Please check

File: app/api/custom/tickets.py [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
     availability = {}
-    tickets = Ticket.query.filter_by(event_id=event_id).all()
+    tickets = Ticket.query.filter_by(event_id=event_id, deleted_at=None).all()
     for i in tickets:
         total_count = i.quantity - get_sold_and_reserved_tickets_count(i.id)
         availability["id"] = i.id
         availability["name"] = i.name
         availability["quantity"] = i.quantity
-        availability["available"] = total_count
+        availability["available"] = total_count > 0 and total_count or 0
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:19]: min(0, total_count)

File: app/api/custom/tickets.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from app.api.attendees import get_sold_and_reserved_tickets_count
+from app.models.ticket import Ticket
+
+tickets_routes = Blueprint('tickets_routes', __name__, url_prefix='/v1/events')
+
+
+@tickets_routes.route('/<int:event_id>/tickets/availability')
+def get_stock(event_id):
+    stock = []
+    availability = {}
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:12]: Move inside loop

File: app/api/custom/tickets.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    tickets = Ticket.query.filter_by(event_id=event_id, deleted_at=None).all()
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:13]: Hidden tickets are also shown

File: app/api/custom/tickets.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    for i in tickets:
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:14]: What is i? Should be `ticket`

File: app/api/custom/tickets.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        total_count = i.quantity - get_sold_and_reserved_tickets_count(i.id)
+        availability["id"] = i.id
+        availability["name"] = i.name
+        availability["quantity"] = i.quantity
+        availability["available"] = total_count > 0 and total_count or 0
+        stock.append(availability.copy())
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:20]: No need to copy

File: app/api/custom/tickets.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from flask import Blueprint, jsonify
+
+from app.api.attendees import get_sold_and_reserved_tickets_count
+from app.models.ticket import Ticket
+
+tickets_routes = Blueprint('tickets_routes', __name__, url_prefix='/v1/events')
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py:6]: Instead of new file, move this to api/tickets

Comment (MeghalBisht) [reply app/api/custom/tickets.py:6]: Okay

File: app/api/custom/tickets.py (hunk 1 above)
Comment (MeghalBisht) [reply app/api/custom/tickets.py:19]: you mean max(0, total_count) ?

File: app/api/custom/tickets.py (hunk 6 above)
Comment (MeghalBisht) [reply app/api/custom/tickets.py]: Please check now, I am not sure if this is the required result.

File: app/api/tickets.py [hunk 7]
```diff
@@ -18,6 +20,26 @@
 from app.models.ticket import Ticket, TicketTag, ticket_tags_table
 from app.models.ticket_holder import TicketHolder
 
+tickets_routes = Blueprint('tickets_routes', __name__, url_prefix='/v1/events')
+
+
+@tickets_routes.route('/<int:event_id>/tickets/availability')
```
Comment (iamareebjamal) [inline app/api/tickets.py:26]: Great. Please make it work with event identifier as well

**Comment by MeghalBisht** [comment]
JSON data is not returned correctly instead an error is thrown, however on logging, the output looks good.
@iamareebjamal Please check once, I'm not able to figure out.

**Log message shows correct data :**

zz

**Error :**

z

**Comment by iamareebjamal** [comment]
It was fine before, this is completely wrong

**Comment by MeghalBisht** [comment]
Okay I will revert, but I cannot complete the event identifier part, please provide an approach

**Comment by iamareebjamal** [comment]
See has-streams API

File: app/api/tickets.py [hunk 8]
```diff
@@ -18,6 +20,36 @@
 from app.models.ticket import Ticket, TicketTag, ticket_tags_table
 from app.models.ticket_holder import TicketHolder
 
+tickets_routes = Blueprint('tickets_routes', __name__, url_prefix='/v1/events')
+
+
+@tickets_routes.route('/<id>/tickets/availability')
+def get_stock(id):
+
+    event = None
```
Comment (iamareebjamal) [inline app/api/tickets.py:29]: No need

Comment (iamareebjamal) [inline app/api/tickets.py:29]:
```suggestion
```

File: app/api/tickets.py [hunk 9]
```diff
@@ -18,6 +20,36 @@
...
+    event_id = None
```
Comment (iamareebjamal) [inline app/api/tickets.py:30]:
```suggestion
    event_id = id
```

File: app/api/tickets.py [hunk 10]
```diff
@@ -18,6 +20,36 @@
...
+
+    if id.isnumeric():
+        event_id = id
+    else:
```
Comment (iamareebjamal) [inline app/api/tickets.py:34]:
```suggestion
    if not id.isnumeric():
```

File: app/api/tickets.py [hunk 11]
```diff
@@ -18,6 +20,36 @@
...
+        event = Event.query.filter_by(identifier=id).one()
+        event_id = event.id
```
Comment (iamareebjamal) [inline app/api/tickets.py:36]:
```suggestion
        event_id = Event.query.filter_by(identifier=id).one_or_404().id
```

File: app/api/tickets.py [hunk 12]
```diff
@@ -18,6 +20,32 @@
 from app.models.ticket import Ticket, TicketTag, ticket_tags_table
 from app.models.ticket_holder import TicketHolder
 
+tickets_routes = Blueprint('tickets_routes', __name__, url_prefix='/v1/events')
+
+
+@tickets_routes.route('/<id>/tickets/availability')
+def get_stock(id):
+
+    event_id = id
```
Comment (iamareebjamal) [inline app/api/tickets.py:29]:
```suggestion
    event_id = id
```

**Pull request merged**